mypy/checker.py

```diff
@@ -1542,9 +1539,20 @@ def check_reverse_op_method(
                 opt_meta = item.type.metaclass_type
                 if opt_meta is not None:
                     forward_inst = opt_meta
+
+        def has_readable_member(typ: Union[UnionType, Instance], name: str) -> bool:
+            # TODO: Deal with attributes of TupleType etc.
+            if isinstance(typ, Instance):
+                return typ.type.has_readable_member(name)
+            return all(
+                (isinstance(x, UnionType) and has_readable_member(x, name))
+                or (isinstance(x, Instance) and x.type.has_readable_member(name))
+                for x in get_proper_types(typ.relevant_items())
+            )
```

Comment:
IIRC mypyc can generate quite inefficient code for nested functions -- have you checked the performance here? Maybe it would be better to have this as a global function somewhere rather than a nested function?

Reply:
Yeah, I know it may generate inefficient code, but IIUC this code should not be called often. I didn't want to define it outside of the method, because the logic there is highly incomplete, and I don't want people to start accidentally using it, we alredy have a lot of code duplication for checkmember.py logic.

If this will cause regressions in some cases (now we have benchmarks run regularly), it should be easy to fix.

Reply:
Makes sense!